ushahidi/platform-client-mzima
# [USH-1609] Add a notification that unsaved changes to a survey will be lost

**Adding confirmation modal for unsaved survey changes**

Below are some of the changes that I have made;

- The confirmModals only have cancel(outline) and delete(red) buttons but no action button (yellow), so I have added that to match the Figma designs.
- Added functionality to track the states of the fields/tasks, meaning that the popup will only appear if the fields/tasks values are edited/different from the initial values. Also, the survey will only be saved if the name field is not empty.
- If a user clicks on 'discard changes', the view will navigate back to the list of surveys. If they click 'save changes', the survey will be saved accordingly.
- Localized the strings

How to test;

1. Open Settings>Surveys>Add Survey and fill in the forms. Try to hit the cancel button and confirm if the popup appears and behaves accordingly.
2. Similarly, select any survey, change any form values/fields, hit the cancel button and confirm if the popup appears and behaves accordingly.
